tjdbc/src/main/java/ru/tinkoff/zeppelin/tjdbc/JDBCInterpreter.java

```diff
@@ -0,0 +1,892 @@
+      if (!isEmpty(password)) {
+        basePropertiesMap.get(propertyKey).setProperty(PASSWORD_KEY, password);
+      }
+    }
+    jdbcUserConfigurations.setPropertyMap(propertyKey, basePropertiesMap.get(propertyKey));
+    if (existAccountInBaseProperty(propertyKey)) {
+      return;
+    }
+    jdbcUserConfigurations.cleanUserProperty(propertyKey);
+
+    String entityName = getEntityName(interpreterContext.getReplName());
+    Optional<UsernamePassword> optUsernamePassword
+            = getUsernamePassword(interpreterContext, entityName);
+    if (optUsernamePassword.isPresent()) {
+      jdbcUserConfigurations.setUserProperty(propertyKey, optUsernamePassword.get());
+    } else {
+      closeDBPool(user, propertyKey);
+    }
+  }
+
+  private void createConnectionPool(String url,
+                                    String user,
+                                    String propertyKey,
+                                    Properties properties)
+          throws SQLException, ClassNotFoundException {
+    ConnectionFactory connectionFactory =
+            new DriverManagerConnectionFactory(url, properties);
+
+    PoolableConnectionFactory poolableConnectionFactory =
+            new PoolableConnectionFactory(connectionFactory, null);
+
+    final String maxConnectionLifetime =
+        defaultIfEmpty(getProperty("zeppelin.jdbc.maxConnLifetime"), "-1");
+
+    poolableConnectionFactory.setMaxConnLifetimeMillis(Long.parseLong(maxConnectionLifetime));
+
+    ObjectPool connectionPool = new GenericObjectPool(poolableConnectionFactory);
+
+    poolableConnectionFactory.setPool(connectionPool);
+    Class.forName(properties.getProperty(DRIVER_KEY));
+    PoolingDriver driver = new PoolingDriver();
+    driver.registerPool(propertyKey + user, connectionPool);
+    getJDBCConfiguration(user).saveDBDriverPool(propertyKey, driver);
+  }
+
+  private Connection getConnectionFromPool(String url,
+                                           String user,
+                                           String propertyKey,
+                                           Properties properties)
+          throws SQLException, ClassNotFoundException {
+    String jdbcDriver = getJDBCDriverName(user, propertyKey);
+    if (!getJDBCConfiguration(user).isConnectionInDBDriverPool(propertyKey)) {
+      createConnectionPool(url, user, propertyKey, properties);
+    }
+    return DriverManager.getConnection(jdbcDriver);
+  }
+
+  private Optional<Connection> establishConnection(InterpreterContext interpreterContext,
+                                                   String propertyKey) {
+    try {
+      return Optional.ofNullable(getConnection(interpreterContext, propertyKey));
+    } catch (InterpreterException e) {
+      throw new IllegalStateException(e.getCause());
+    } catch (IOException | SQLException | ClassNotFoundException e) {
+      logger.warn("Database connection cannot be established");
```

Comment:
```suggestion
      logger.warn("Database connection cannot be established", e);
```
Иначе по логам никогда не поймём почему "connection cannot be established".

Reply:
OK